### PR TITLE
Expose stream parameter to public rolling APIs

### PR DESCRIPTION
**Pull request by srinivasyadav18**
## Description
Add stream parameter to public rolling APIs.

- `rolling()`

- `grouped_rolling_window()`
- `grouped_time_range_rolling_window()`
- `grouped_range_rolling_window()`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by mythrocks — APPROVED** [review]
Yep, LGTM. Thanks for making this change, @mhaseeb123.

**Review by mhaseeb123 — APPROVED** [review]
LGTM!

File: cpp/tests/CMakeLists.txt [hunk 1]
```diff
@@ -697,6 +697,7 @@ ConfigureTest(STREAM_REPLACE_TEST streams/replace_test.cpp STREAM_MODE testing)
 ConfigureTest(STREAM_SEARCH_TEST streams/search_test.cpp STREAM_MODE testing)
 ConfigureTest(STREAM_SORTING_TEST streams/sorting_test.cpp STREAM_MODE testing)
 ConfigureTest(STREAM_REDUCTION_TEST streams/reduction_test.cpp STREAM_MODE testing)
+ConfigureTest(STREAM_ROLLING_TEST streams/rolling_test.cpp STREAM_MODE testing)
```
Comment (davidwendt) [inline cpp/tests/CMakeLists.txt:700]: Could you update these 2 lines to move them alphabetically in the list here?

**Comment by vyasr** [comment]
@srinivasyadav18 could you take a look at the failing tests here?

**Comment by srinivasyadav18** [comment]
Hi @vyasr . Yes, I found out what's causing the issue. The **scalar's** used by the `grouped_rolling_window` **algorithm** are not using the actual `stream` parameter, instead they are using the `default stream`. The issue is propogating from [here](https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/src/rolling/range_window_bounds.cpp#L33-L62). I will try to debug and fix it soon.

**Comment by vyasr** [comment]
Nice find!

**Pull request merged**